src/scenes/App/components/TermsAndConditions/index.jsx

```diff
@@ -0,0 +1,274 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { withStyles } from 'material-ui/styles';
+import Paper from 'material-ui/Paper';
+import Typography from 'material-ui/Typography';
+import { FormattedMessage } from 'react-intl';
+import Grid from 'material-ui/Grid';
+import { FormGroup, FormControlLabel } from 'material-ui/Form';
+import Checkbox from 'material-ui/Checkbox';
+import Button from 'material-ui/Button';
+
+import AppConfig from '../../../../config/app';
+import styles from './styles';
+
+const TNC_ACCEPTED = 'tncAccepted';
+
+@withStyles(styles, { withTheme: true })
+export default class TermsAndConditions extends React.PureComponent {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    locale: PropTypes.string.isRequired,
+    langHandler: PropTypes.func,
+  };
+
+  static defaultProps = {
+    langHandler: undefined,
+  };
+
+  state = {
+    checkboxChecked: false,
+  };
+
+  render() {
+    const { classes, locale } = this.props;
+    const { checkboxChecked } = this.state;
+
+    let tncAccepted = localStorage.getItem(TNC_ACCEPTED);
+    tncAccepted = _.isUndefined(tncAccepted) || _.isNull(tncAccepted) ? false : tncAccepted;
+
+    return (
+      <div
+        className={classes.tncBg}
+        style={
+          {
+            opacity: tncAccepted ? 0 : 1,
+            display: tncAccepted ? 'none' : 'block',
+          }
+        }
+      >
+        <Grid container>
+          <Grid item xs>
+            <Button color="primary" variant="raised" className={classes.langBtn} onClick={this.props.langHandler}>
+              <FormattedMessage id="language.select" defaultMessage="中文" />
+            </Button>
+          </Grid>
+          <Grid item xs={8}>
+            <Paper className={classes.tncWrapper}>
+              {this.renderTitle()}
+              {this.renderGeneral()}
+              {this.renderDefinitions()}
+              {this.renderSubordination()}
+              {this.renderParticipate()}
+              {this.renderRepresentations()}
+              {this.renderAwards()}
+              {this.renderPowers()}
+              {this.renderResponsibilities()}
+              {this.renderIp()}
+              {this.renderSupport()}
+              {this.renderMisc()}
+            </Paper>
+            <FormGroup row className={classes.formGroup}>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    checked={checkboxChecked}
+                    value="checkboxChecked"
+                    onChange={this.handleCheckedChange('checkboxChecked')}
```

Comment:
instead of having a `handleCheckedChange` that only get's used once, why not just do 
```
onChange={(e) => this.setState({ checkboxChecked: e.target.value })}
```
or an even cooler way
```
onChange={({ target: { value } }) => this.setState({ checkboxChecked: value })}
```
but the 1st way is shorter, so I'd go w/ that lol